### PR TITLE
Refactoring .net Dockerfile and bash scripts

### PR DESCRIPTION
**Pull request by neomatrix369**
Refactoring Dockerfile for csharp, fsharp and vbnet to build a docker image inherited from a common image when building the individual language containers

### CODE REVIEW

**Comment by neomatrix369** [comment]
```
~~~ Summary of test executions ~~~
  ~~~ Passed Tests ~~~
  language=csharp|challenge=SUM|expected=100|actual=100
  language=csharp|challenge=TST|expected=50|actual=50
  language=csharp|challenge=CHK|expected=0|actual=0
  language=csharp|challenge=xyz|expected=0|actual=0
  4 test(s) passed

  ~~~ Failed Tests ~~~
  0 test(s) failed
```

```
~~~ Summary of test executions ~~~
  ~~~ Passed Tests ~~~
  language=fsharp|challenge=SUM|expected=100|actual=100
  language=fsharp|challenge=TST|expected=50|actual=50
  language=fsharp|challenge=CHK|expected=0|actual=0
  language=fsharp|challenge=xyz|expected=0|actual=0
  4 test(s) passed

  ~~~ Failed Tests ~~~
  0 test(s) failed
```

```
~~~ Summary of test executions ~~~
  ~~~ Passed Tests ~~~
  language=vbnet|challenge=SUM|expected=100|actual=100
  language=vbnet|challenge=TST|expected=50|actual=50
  language=vbnet|challenge=CHK|expected=0|actual=0
  language=vbnet|challenge=xyz|expected=0|actual=0
  4 test(s) passed

  ~~~ Failed Tests ~~~
  0 test(s) failed
```

File: container/buildDockerImage.sh [hunk 1]
```diff
@@ -7,15 +7,29 @@ set -o pipefail
...
+BASE_IMAGE_TAG=${ROOT_BASE_IMAGE_TAG}
 
 function die() { echo >&2 $1; exit 1; }
 [ "$#" -eq 1 ] || die "Usage: $0 LANGUAGE_ID"
-LANGUAGE_ID=$1
 
 echo "~~~~~~ Refreshing base image ~~~~~~"
-docker build -t ${BASE_IMAGE_TAG} "${IMAGES_DIR}/base/."
+if [[ "${BASE}" == "base"  ]]; then
+    docker build -t ${BASE_IMAGE_TAG} "${IMAGES_DIR}/${BASE}/."
```
Comment (julianghionoiu) [inline container/buildDockerImage.sh]: You still need to refresh the base, even for dotnet, as the dotnet-base depends on base.

File: container/buildDockerImage.sh [hunk 2]
```diff
@@ -7,15 +7,29 @@ set -o pipefail
...
 IMAGES_DIR="${SCRIPT_CURRENT_DIR}/images"
 DEFAULT_IMAGE_PREFIX="accelerate-io/dpnt-coverage-"
-BASE_IMAGE_VERSION=$( cat "${IMAGES_DIR}/base/version.txt" | tr -d " " | tr -d "\n" )
-BASE_IMAGE_TAG="${DEFAULT_IMAGE_PREFIX}base:${BASE_IMAGE_VERSION}"
+
+BASE="base"
+LANGUAGE_ID=$1
+case ${LANGUAGE_ID} in
+    csharp|fsharp|vbnet)
+    BASE="dotnet" ;;
```
Comment (julianghionoiu) [inline container/buildDockerImage.sh]: We should rename that to dotnet-base

File: container/images/dotnet/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+RUN ln -s /usr/lib/apt/methods/http /usr/lib/apt/methods/https
+RUN apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 3FA7E0328081BFF6A14DA29AA6A19B38D3D831EF
+RUN apt-get update && apt-get -y install apt-transport-https curl
+RUN echo "deb https://download.mono-project.com/repo/ubuntu stable-jessie main" | tee /etc/apt/sources.list.d/mono-official-stable.list
+
+RUN apt-get update && apt-get install -qy \
+    mono-devel     \
+    mono-complete  \
+    mono-profiler  \
+    nunit-console
```
Comment (julianghionoiu) [inline container/images/dotnet/Dockerfile]: WE should clean up after yum to reduce image size " rm -r /var/lib/apt/lists/*"

**Pull request merged**